pkgs/applications/misc/j4-dmenu-desktop/default.nix

```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchFromGitHub, cmake, dmenu }:
+
+stdenv.mkDerivation rec {
+  name    = "j4-dmenu-desktop-${version}";
+  version = "2.14";
+
+  src = fetchFromGitHub {
+    owner  = "enkore";
+    repo   = "j4-dmenu-desktop";
+    rev    = "r${version}";
+    sha256 = "14srrkz4qx8qplgrrjv38ri4pnkxaxaq6jy89j13xhna492bq128";
+  };
+
+  patchPhase = ''
+    substituteInPlace ./src/Main.hh \
+      --replace "\"dmenu -i\"" "\"${dmenu}/bin/dmenu -i\""
+  '';
+
+  buildInputs = [ cmake dmenu ];
```

Comment:
To be nitpicky, `cmake` could be moved to `nativeBuildInputs` as its a build-time only dependency, and I don't think you actually need to add `dmenu` to inputs at all (unless the build somehow expects it to be in the environment, that is; if you all you need it for is to write a reference to it into the patched file, that is sufficient to ensure that it ends up in the closure).